src/main/java/io/cdap/plugin/dlp/SensitiveRecordDecrypt.java

```diff
@@ -0,0 +1,225 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Class for the Redact DLP transform plugin
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name(SensitiveRecordDecrypt.NAME)
+@Description(SensitiveRecordDecrypt.DESCRIPTION)
+public class SensitiveRecordDecrypt extends Transform<StructuredRecord, StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SensitiveRecordDecrypt.class);
+  public static final String NAME = "SensitiveRecordDecrypt";
+  public static final String DESCRIPTION = "SensitiveRecordDecrypt";
+
+  private StageMetrics metrics;
+
+  // Stores the configuration passed to this class from user.
+  private final Config config;
+
+  // DLP service client for managing interactions with DLP service.
+  private DlpServiceClient client;
+
+  // Required fields that need to be sent to DLP for the transform to work, this variable is used to cache the set
+  // since it will not change during the execution of the plugin
+  private Set<String> requiredFields = null;
+
+  @VisibleForTesting
+  public SensitiveRecordDecrypt(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    config.validate(stageConfigurer.getFailureCollector(), stageConfigurer.getInputSchema());
+
+    stageConfigurer.setOutputSchema(stageConfigurer.getInputSchema());
+    super.configurePipeline(pipelineConfigurer);
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    metrics = context.getMetrics();
+    client = DlpServiceClient.create(getSettings());
+    try {
+      requiredFields = config.getRequiredFields();
+    } catch (Exception e) {
+      LOG.warn("Unable to get list of required fields, defaulting to an empty set.", e);
+      requiredFields = new HashSet<>();
+    }
+  }
+
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    config.validate(context.getFailureCollector(), context.getInputSchema());
+    context.getFailureCollector().getOrThrowException();
+    if (config.customTemplateEnabled) {
+      String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+      GetInspectTemplateRequest request = GetInspectTemplateRequest.newBuilder().setName(templateName).build();
+
+      try {
+        if (client == null) {
+          client = DlpServiceClient.create(getSettings());
+        }
+        InspectTemplate template = client.getInspectTemplate(request);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(
+          "Unable to validate template name. Ensure template ID matches the specified ID in DLP");
```

Comment:
Can we put a link to were they can check the Id match

Reply:
done